interfaces/snap_app_set.go

```diff
@@ -135,6 +135,70 @@ func (a *SnapAppSet) SlotLabelExpression(slot *ConnectedSlot) string {
 	return labelExpr(apps, hooks, info)
 }
 
+// RunnableType is an enumeration of the different types of runnables that can
+// be present in a snap.
+type RunnableType int
+
+const (
+	RunnableApp RunnableType = iota
+	RunnableHook
+	RunnableComponentHook
+)
+
+// Runnable represents a runnable element of a snap.
+type Runnable struct {
+	// CommandName is the name of the command that is run when this runnable
+	// runs.
+	CommandName string
+	// SecurityTag is the security tag associated with the runnable. Security
+	// tags are used by various security subsystems as "profile names" and
+	// sometimes also as a part of the file name.
+	SecurityTag string
+	// Type is the type of the runnable, either an app, a hook, or a component
+	// hook.
+	Type RunnableType
```

Comment:
question: is Type actually used so far?

Reply:
It is used to group the returned slice from `.Runnables()`, but we could get away without it if desired. I think at some point it had a greater purpose, but ended up not being needed.